### PR TITLE
feat: detect and support window.ipfs.enable experiment

### PR DESCRIPTION
**Pull request by lidel**
Just to get the ball rolling, this PR adds a simple heuristic for forward-compatible detection of IPFS API exposed by experimental branch of ipfs-companion: https://github.com/ipfs-shipyard/ipfs-companion/issues/589

@alanshaw I can take care of releasing this if you make me a colaborator on NPM.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -2,7 +2,15 @@ module.exports = function getIpfs (opts) {
   opts = opts || {}
 
   return new Promise(function (resolve, reject) {
-    if (window.ipfs) return resolve(window.ipfs)
+    if (window.ipfs) {
+      // forward-compatible migration
+      // https://github.com/ipfs-shipyard/ipfs-companion/issues/589
+      if (typeof window.ipfs.enable === 'function') {
+        return resolve(window.ipfs.enable())
```
Comment (alanshaw) [inline index.js]: Pass wanted permissions...so we don't have to do another release??

```suggestion
        return resolve(window.ipfs.enable(opts.permissions))
```

**Review by alanshaw — CHANGES_REQUESTED** [review]
Code LGTM - would you mind adding a test?

**Comment by lidel** [comment]
@alanshaw done, check now:
- added test
- added support for overriding `defaults`
  - did not want to name it 'permissions` as we may want to pass other flags

**Comment by alanshaw** [comment]
> - added support for overriding `defaults`

hmm, I don't expect to pass "defaults" normally! It suggests something is going to override them...when actually what I pass would usually override anything set by default.

>   - did not want to name it 'permissions` as we may want to pass other flags 

Unless you have something in mind right now I'd suggest sticking to `permissions` (we know we want that) and change it when we want to pass something else/more.

**Comment by lidel** [comment]
@alanshaw lets go with `permissions` for now (https://github.com/ipfs-shipyard/window.ipfs-fallback/pull/2/commits/bed7c7206f0dd8ffaee18ada8f5bfd4bc49cab03)

**Review by alanshaw — APPROVED** [review]

**Pull request merged**